src/settings/toml/manifest.rs

```diff
@@ -512,6 +513,96 @@ impl Manifest {
     }
 }
 
+#[derive(Debug, Clone, Default, PartialEq)]
+pub struct LazyAccountId(OnceCell<String>);
+
+impl Serialize for LazyAccountId {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        self.0.get().serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for LazyAccountId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        Option::deserialize(deserializer).map(Into::into)
+    }
+}
```

Comment:
This seems like something I should be able to do with the serde derive macros, but I'm not sure what the syntax is.

Reply:
I think it could be if not for the `OnceCell` wrapper. I _think_ serde supports newtype / tuple structs via derive.

Reply:
This a moot point now that the deserialize impl handles the empty string, I don't see any way that could be a derive.